File: test/support/classifier.py
from vowpalwabbit.sklearn_vw import VWClassifier

X = [[1, 2], [3, 4], [5, 6], [7, 8]]
y = [-1, -1, 1, 1]

model = VWClassifier(loss_function='logistic', l=0.01, l2=0.1)
model.fit(X, y)

print(model.predict(X))
print(model.score(X, y))

File: lib/vowpalwabbit/classifier.rb
module VowpalWabbit
  class Classifier < Model
    def initialize(**params)
      super(loss_function: "logistic", **params)
    end

    def predict(x)
      predictions = super
      predictions.map { |v| v >= 0 ? 1 : -1 }
    end

    def score(x, y = nil)
      y_pred, y = predict_for_score(x, y)
      y_pred.map! { |v| v >= 0 ? 1 : -1 }
      y_pred.zip(y).select { |yp, yt| yp == yt }.count / y.count.to_f
    end
  end
end
